src/core/ext/filters/client_channel/resolver/dns/c_ares/dns_resolver_ares.c

```diff
@@ -204,7 +204,7 @@ static char *choose_service_config(char *service_config_choice_json) {
         int random_pct = rand() % 100;
         int percentage;
         if (sscanf(field->value, "%d", &percentage) != 1 ||
-            random_pct > percentage) {
+            random_pct > percentage || percentage == 0) {
```

Comment:
I'm not sure this is necessary.  The percentage is an optional field, and if you don't want to restrict by percentage, you should just not specify the field.  If you do explicitly specify a value of 0, it's technically correct that no clients should select it.

What test flake is being caused by this?  It's not clear to me why this would cause any flakes.

Reply:
The flakiness was seen on a test that uses a value of zero as the percentage. In that case, if 
`random_pct` turns out to be zero, then this conditional doesn't get taken and the service config will be unexpectedly picked.

FTR it could be repro'd on the branch of that PR (before it's most recent commit), but running the test server:

```
test/cpp/naming/test_dns_server.py --port=15353 --records_config_path=test/cpp/naming/resolver_test_record_groups.yaml
```

and then running a test that exercises a value of zero as a percentage:

```
CONFIG=dbg make resolver_component_test
while [[ $? == 0 ]]; do GRPC_DNS_RESOLVER=ares  bins/dbg/resolver_component_test --target_name=ipv4-cpp-config-has-zero-percentage.resolver-tests.grpctestingexp. --expected_addrs=1.2.3.4:443,False --expected_chosen_service_config= --expected_lb_policy= --local_dns_server_address=127.0.0.1:15353; done;
```

Reply:
Thanks for the explanation.  You're right, this is the right fix.